routers/api/v1/repo/pull.go

```diff
@@ -834,12 +847,17 @@ func MergePullRequest(ctx *context.APIContext) {
 		ctx.Error(http.StatusInternalServerError, "Merge", err)
 		return
 	}
+	// commit dbCtx
+	if err := committer.Commit(); err != nil {
+		ctx.InternalServerError(err)
+		return
+	}
```

Comment:
```suggestion
	// commit dbCtx
	if err := committer.Commit(); err != nil {
		ctx.InternalServerError(err)
		return
	}
	committer = nil
	dbCtx = nil
```

The logic is too complex here, it's better to reset these variables to prevent from misuse.